lib/spack/spack/environment/environment.py

```diff
@@ -2668,6 +2668,26 @@ def __init__(self, manifest_dir: Union[pathlib.Path, str]) -> None:
         self.yaml_content = with_defaults_added
         self.changed = False
 
+    def _all_matches(self, user_spec: str) -> List[str]:
+        """Maps the input string to the first equivalent user spec in the manifest,
+        and returns it.
+
+        Args:
+            user_spec: user spec to be found
+
+        Raises:
+            ValueError: if no equivalent match is found
+        """
+        result = []
+        for yaml_spec_str in self.pristine_configuration["specs"]:
+            if Spec(yaml_spec_str) == Spec(user_spec):
```

Comment:
```suggestion
            if Spec(yaml_spec_str).satisifies(user_spec)):
```

Reply:
Can we discuss it for a behavioral change later? This https://github.com/spack/spack/issues/39387#issuecomment-1674303464 still holds - i.e. currently we can remove the middle spec using exact match:
```yaml
spack:
  specs:
  - bricks+foo
  - bricks
  - bricks~bar
```
With `satisfies` we can't

Reply:
What we could do is add an additional argument to `spack remove` to specify the semantic of matches, but I'm pretty convinced that just substituting `satisfies` is wrong in this context.

Reply:
Can we move this discussion to another PR? Or implement this in a way that doesn't change the semantic?